### ISSUE TITLE
Add Tuple handling

### ISSUE THREAD

**Issue by nicolashery** [paf31/purescript-foreign-generic#6]
I have a piece of data that's represented in JS-land as:

```javascript
var business = {
  name: "O'Brien's Irish Pub & Restaurant",
  category: ["Irish Pub", "irish-pub"]
};
```

Which I'd like to represent in PureScript as:

```purescript
newtype Business = Business {
  name :: String,
  category :: Tuple String String
}
```

Seems like a common enough use case to add support for `Tuple` and be able to use generics for such data structure?

**Comment by paf31** [comment]
:+1: Sounds good.

**Comment by paf31** [comment]
Fixed in 0.3.0.
